lib/boutiques_bids_subject_subsetter.rb

```diff
@@ -82,42 +82,43 @@ module BoutiquesBidsSubjectSubsetter
   def descriptor_with_special_input(descriptor)
-    descriptor = descriptor.dup
-
-    # Add new input
-    new_input = BoutiquesSupport::Input.new(
-      "name"          => "Partial path for BIDS sub setter",
-      "id"            => "#{InputIdForBidsSubjectSubsetter}",
-      "description"   => "List of files to keep within the BIDS subject. The filtering will take care of keeping the associate file together.\n If 'anat/sub-123456_ses-V01_acq-anat_run-1_TB1TFL.nii.gz' the associated *.json file will be kept too.\n If files are not specified for some subfolder no filtering will be perfomed.\n That should include the folder name and at list the file name with or without extension (e.g: 'anat/T1')",
-      "type"          => "String",
-      "optional"      => true,
-      "list"          => true
-    )
-
-    descriptor.inputs <<= new_input
-
-    # Add new group with that input
-    groups       = descriptor.groups || []
-    cb_mod_group = groups.detect { |group| group.id == 'cbrain_bids_extensions' }
-    if cb_mod_group.blank?
-      cb_mod_group = BoutiquesSupport::Group.new(
-        "name"        => 'CBRAIN BIDS Extensions',
-        "id"          => 'cbrain_bids_extensions',
-        "description" => 'Special options for BIDS data files',
-        "members"     => [],
+    descriptor       = descriptor.dup
+    map_ids_for_bbss = descriptor.custom_module_info('BoutiquesBidsSubjectSubsetter')
+    groups           = descriptor.groups || []
+
+    map_ids_for_bbss.each do |dir_input_id, keep_input_id|
+      dir_input_name = descriptor.input_by_id(dir_input_id).name
+
+      new_input = BoutiquesSupport::Input.new(
+        "name"          => "Partial path for BIDS sub setter",
+        "id"            => "#{keep_input_id}",
+        "description"   => "List of files to keep within the BIDS subject define by '#{dir_input_name}'. The filtering will take care of keeping the associate file together.\n If 'anat/sub-123456_ses-V01_acq-anat_run-1_TB1TFL.nii.gz' the associated *.json file will be kept too.\n If files are not specified for some subfolder no filtering will be perfomed.\n That should include the folder name and at list the file name with or without extension (e.g: 'anat/T1')",
+        "type"          => "String",
+        "optional"      => true,
+        "list"          => true
       )
-      groups.unshift cb_mod_group
-    end
 
-    cb_mod_group.members <<= new_input.id
+      descriptor.inputs <<= new_input
+
+      # Add new group with that input
+      cb_mod_group = groups.detect { |group| group.id == 'cbrain_bids_extensions' }
+      if cb_mod_group.blank?
+        cb_mod_group = BoutiquesSupport::Group.new(
+          "name"        => 'CBRAIN BIDS Extensions',
+          "id"          => 'cbrain_bids_extensions',
+          "description" => 'Special options for BIDS data files',
+          "members"     => [],
+        )
+        groups.unshift cb_mod_group
+      end
+
+      cb_mod_group.members <<= new_input.id
```

Comment:
Alhtough `<<=` works, it's supposed to be `<<` ; You should push to the existing array, not push and re-assign the array.